dev-tools/support-bundle-reader-lib/src/bundle_accessor.rs

```diff
@@ -0,0 +1,260 @@
+    ) -> Result<BoxedFileAccessor<'a>>
+    where
+        Self: 'a;
+}
+
+pub struct StreamedFile<'a> {
+    client: &'a nexus_client::Client,
+    id: SupportBundleUuid,
+    path: Utf8PathBuf,
+    stream: Option<Pin<Box<dyn Stream<Item = reqwest::Result<Bytes>> + Send>>>,
+    buffer: Bytes,
+}
+
+impl<'a> StreamedFile<'a> {
+    fn new(
+        client: &'a nexus_client::Client,
+        id: SupportBundleUuid,
+        path: Utf8PathBuf,
+    ) -> Self {
+        Self { client, id, path, stream: None, buffer: Bytes::new() }
+    }
+
+    // NOTE: This is a distinct method from "new", because ideally some day we could
+    // use range requests to stream out portions of the file.
+    //
+    // This means that we would potentially want to restart the stream with a different position.
+    async fn start_stream(&mut self) -> Result<()> {
+        // TODO: Add range headers, for range requests? Though this
+        // will require adding support to Progenitor + Nexus too.
+        let stream = self
+            .client
+            .support_bundle_download_file(
+                self.id.as_untyped_uuid(),
+                self.path.as_str(),
+            )
+            .await
+            .with_context(|| {
+                format!(
+                    "downloading support bundle file {}: {}",
+                    self.id, self.path
+                )
+            })?
+            .into_inner_stream();
+
+        self.stream = Some(Box::pin(stream));
+        Ok(())
+    }
+}
+
+impl AsyncRead for StreamedFile<'_> {
+    fn poll_read(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<io::Result<()>> {
+        while self.buffer.is_empty() {
+            match futures::ready!(
+                self.stream
+                    .as_mut()
+                    .expect("Stream must be initialized before polling")
+                    .as_mut()
+                    .poll_next(cx)
+            ) {
+                Some(Ok(bytes)) => {
+                    self.buffer = bytes;
+                }
+                Some(Err(e)) => {
+                    return Poll::Ready(Err(io::Error::new(
+                        io::ErrorKind::Other,
+                        e,
+                    )));
+                }
+                None => return Poll::Ready(Ok(())), // EOF
+            }
+        }
+
+        let to_copy = std::cmp::min(self.buffer.len(), buf.remaining());
+        buf.put_slice(&self.buffer[..to_copy]);
+        self.buffer.advance(to_copy);
+
+        Poll::Ready(Ok(()))
+    }
+}
+
+/// Access to a support bundle from the internal API
+pub struct InternalApiAccess<'a> {
+    client: &'a nexus_client::Client,
+    id: SupportBundleUuid,
+}
+
+impl<'a> InternalApiAccess<'a> {
+    pub fn new(
+        client: &'a nexus_client::Client,
+        id: SupportBundleUuid,
+    ) -> Self {
+        Self { client, id }
+    }
+}
+
+// Access for: The nexus internal API
```

Comment:
I might need to refactor this crate a bit, but I'm expecting we'll also add one of these for `ExternalApiAccess` if we want to re-use this crate in the CLI

Reply:
The need for `ExternalApiAccess` would be a tool consuming the support-bundle-reader-lib that a customer could use rather than omdb?